packages/apps/plugins/plugin-sheet/src/components/Toolbar/Toolbar.tsx

```diff
@@ -0,0 +1,175 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import {
+  type Icon,
+  ChatText,
+  TextAlignCenter,
+  TextAlignLeft,
+  TextAlignRight,
+  TextStrikethrough,
+  TextB,
+} from '@phosphor-icons/react';
+import { createContext } from '@radix-ui/react-context';
+import React, { type PropsWithChildren } from 'react';
+
+import {
+  DensityProvider,
+  ElevationProvider,
+  type ThemedClassName,
+  Toolbar as NaturalToolbar,
+  useTranslation,
+} from '@dxos/react-ui';
+
+import { ToolbarButton, ToolbarSeparator, ToolbarToggleButton } from './common';
+import { SHEET_PLUGIN } from '../../meta';
+import { type Formatting } from '../../types';
+
+//
+// Root
+//
+
+export type ToolbarAction = {
+  type: string;
+};
+
+export type ToolbarProps = ThemedClassName<
+  PropsWithChildren<{
+    onAction?: ({ type }: ToolbarAction) => void;
+  }>
+>;
+
+const [ToolbarContextProvider, useToolbarContext] = createContext<ToolbarProps>('Toolbar');
+
+const ToolbarRoot = ({ children, onAction, classNames }: ToolbarProps) => {
+  return (
+    <ToolbarContextProvider onAction={onAction}>
+      <DensityProvider density='fine'>
+        <ElevationProvider elevation='chrome'>
+          <NaturalToolbar.Root classNames={['is-full shrink-0 overflow-x-auto overflow-y-hidden p-1', classNames]}>
+            {children}
+          </NaturalToolbar.Root>
+        </ElevationProvider>
+      </DensityProvider>
+    </ToolbarContextProvider>
+  );
+};
+
+// TODO(burdon): Generalize.
+type ButtonProps = {
+  type: string;
+  Icon: Icon;
+  getState: (state: Formatting) => boolean;
+  disabled?: (state: Formatting) => boolean;
+};
+
+//
+// Alignment
+//
+
+// TODO(burdon): Detect and display current state.
+const alignmentOptions: ButtonProps[] = [
+  { type: 'left', Icon: TextAlignLeft, getState: (state) => false },
+  { type: 'center', Icon: TextAlignCenter, getState: (state) => false },
+  { type: 'right', Icon: TextAlignRight, getState: (state) => false },
```

Comment:
Passing icon components will soon be deprecated. Components should begin to use `iconSymbol` strings and render an `<svg><use href={`/icons.svg#{iconSymbol}`}/></svg>` (we should add a component for this).

These icons would be `ph--text-align-left--regular`, `ph--text-align-center--regular`, and `ph--text-align-right--regular`.

Reply:
ok cool -- that's close to what i'm doing via `twind` in the edge monorepo (purely server side).